Source/DataSources/ModelVisualizer.js

```diff
@@ -150,6 +154,43 @@ define([
     };
 
     /**
+     * Computes a bounding sphere which encloses the visualization produced for the specified entity.
+     *
+     * @param {Entity} entity The entity whose bounding sphere to compute.
+     * @param {BoundingSphere} result The bounding sphere onto which to store the result.
+     * @returns {BoundingSphereState} BoundingSphereState.DONE if the result contains the bounding sphere,
+     *                       BoundingSphereState.PENDING if the result is still being computed, or
+     *                       BoundingSphereState.FAILED if the entity has no visualization in the current scene.
+     */
+    ModelVisualizer.prototype.getBoundingSphere = function(entity, result) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(entity)) {
+            throw new DeveloperError('entity is required.');
+        }
+        if (!defined(result)) {
+            throw new DeveloperError('result is required.');
+        }
+        //>>includeEnd('debug');
+
+        var modelData = this._modelHash[entity.id];
+        if (!defined(modelData)) {
+            return BoundingSphereState.FAILED;
+        }
+
+        var model = modelData.modelPrimitive;
+        if (!defined(model) || !model.show) {
+            return BoundingSphereState.FAILED;
+        }
+
+        if (!model.ready) {
+            return BoundingSphereState.PENDING;
+        }
+
+        BoundingSphere.transform(model.boundingSphere, model.modelMatrix, result);
```

Comment:
I know exactly what the problem is, I just didn't make any changes because I know you wanted to look at it, I'll open a separate PR (against master) with the full explanation and you guys can look at it whenever you have time